spec/features/pictures_feature_spec.rb

```diff
@@ -0,0 +1,154 @@
+require 'rails_helper'
+
+feature 'Pictures' do
+  # As a user
+  # So that I can use Instagram
+  # I would like to post pictures on it
+
+  context 'no pictures have been posted' do
+    scenario 'should show a button to post a picture' do
+      visit '/pictures'
+      expect(page).to have_link('Post a picture')
+    end
+  end
+
+  context 'posting pictures' do
+    context 'when signed in' do
+      before(:each) do
+        user = FactoryGirl.create(:user, :first)
+        sign_in(user.email, user.password)
+      end
+
+      scenario 'user can upload a picture' do
+        visit '/pictures/'
+        click_link 'Post a picture'
+        fill_in 'Description', with: 'My first picture!'
+        page.attach_file('Image', Rails.root + 'spec/factories/test.jpg')
+        expect(attach_file('Image', 'spec/factories/test.jpg')).to be_truthy
+      end
+
+      scenario 'throws error if user does not upload an image' do
+        click_link 'Post a picture'
+        fill_in 'Description', with: 'My first picture!'
+        click_button 'Post'
+        expect(page).to have_content('Error: You must upload a picture')
+      end
+    end
+
+    context 'when signed out' do
+      scenario 'user cannot post pictures' do
+        visit '/'
+        click_link 'Post a picture'
+        expect(page).to have_content 'Log in'
+      end
+    end
+  end
+
+  # As a user
+  # So that I can see pictures posted by other users
+  # I would like to see all users' pictures on the homepage
+  context 'viewing pictures' do
+    before(:each) do
+      user = FactoryGirl.create(:user, :first)
+      sign_in(user.email, user.password)
+    end
+
+    scenario 'pictures can be viewed' do
+      visit '/pictures/new'
+      fill_in 'Description', with: 'My first picture!'
```

Comment:
`fill_in 'Description', with: 'My first picture!'` has been repeated a few times at this point so consider refactoring it to a helper method, `#fill_in_description`.